### ISSUE TITLE
print(...) + input() does not work as expected

### ISSUE THREAD

**Issue by malvill** [pyscript/pyscript#2007]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I already searched in Google and didn't find any good information or help

### What happened?

Running the following Python code in pyscript terminal
```python
print('anything')
a = input()
```
 results as 
 ```
anything
anything
```
 So whenever an empty input is placed after print(), the argument of print function is logged twice

### Console info

```shell
no console errors
```

### Additional Context

version 2024.3.1

**Comment by malvill** [comment]
sorry, here is the document that I insert via iframe
`<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>PyProject</title>
    <link rel="stylesheet" href="https://pyscript.net/releases/2024.3.1/core.css">
    <script type="module" src="https://pyscript.net/releases/2024.3.1/core.js"></script>
  </head>
  <body>
    <script type="py" terminal worker>
          print('enter your name: ')
          name = input('')
          print(f'Hello, {name}')
    </script>
  </body>
</html>`

Using input('') instead of input() does not help. And answering your question, no douple print if the order is inverted

**Comment by WebReflection** [comment]
look closer ... `input('​')` is **not** `input('')` ... there's a zero width char in the former ... the length or size of `input('') ` is 9, the length or size of my hint, `input('​')` is 10.

You really need to copy and paste this:

```
input('​')
```

**Comment by WebReflection** [comment]
I've just tried this and it works wonderfully ... we should probably fix this on our side by default (no string or empty string) but the issue is upstream:

```html
<!doctype html>
<html lang="en">
    <head>
        <meta charset="UTF-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
        <link rel="stylesheet" href="https://pyscript.net/releases/2024.3.2/core.css">
        <script type="module" src="https://pyscript.net/releases/2024.3.2/core.js"></script>
    </head>
    <body>
        <script type="py" terminal worker>
            print('enter your name')
            a = input('​')
            print(f'Hello, {a}')
        </script>
    </body>
</html>
```

**Comment by WebReflection** [comment]
so ... we discussed this during the community call (live and available on discord: https://discord.com/channels/972017612454232116/972017612454232119/1222232910762086551)

usually, when you ask something to a user, even on the Web via `prompt`, you actually ask something to the user ... that is:

```python
a = input('enter your name: ')
print(f'Hello, {a}')
```

so that while it should be possible to ask for any input in a new empty line, it's not clear why one would need that, but that zero-with-space is a work around for the requirement, something I don't think we shuold put by default into the I/O of the program, so that my early atempt to fix it behind the upstream issue, is actually both not really working, and not the solution to this issue.

In short: are you OK in using `input(your_prompt)` the wasy it's meant to be used on Python, the Web, and other terminal like scenarios? If **no** is the answer, please use the workaround, otherwise I hope this answers your initial issue.

**Comment by malvill** [comment]
Thank you so much for your fast and thorough reply! My app is supposed to run other people's code which I can't control, so I guess I'll need to use a workaround then.

**Comment by WebReflection** [comment]
not sure if it helps, but before any evaluation you could replace any `input()` with `input('​')` (zero-width-space chare in there) and call it a day, if you think your code is unpredictable around this topic ... you deal with Python code injection and you have all rights to author it, when/if needed.
